### DISCUSSION TITLE
Did anyone successfully logged in using Instagram provider?

### DISCUSSION THREAD

**Discussion by laszlo1337** [pocketbase/pocketbase#4171] [Q&A]
I don't understand any part of their genious way of doing things. 
I Implemented Google oauth and Discord oauth without any issue at all, it's simple and straightforwad. 
Even GPT4 does not know how to implement Instagram oauth (it says to use "Basic Display" but note on basic display page says it cannot be used for auth). Anyway I tried and pocketbase is just crashing when provided with key/secret from there.

The whole thing makes no sense at all. Anyone had success? I need auth by instagram because that's where product will be advertised, so makes sense to do "continue with instagram" button

**Comment by ganigeorgiev** [comment]
> pocketbase is just crashing when provided with key/secret from there

I don't understand what do you mean by "just crash" but this doesn't seem right.

Try to start the executable with `--dev` and/or change the "Min log level" to `-4` in the _Admin UI > Logs > Logs settings_ popup and check whether there is any more detailed error message in the console or in the Logs dashboard.

**Reply by laszlo1337** [reply]
I'm sorry I was supposed to write that my app crashes, as it receives 401 during auth-refresh request. When I disable Instagram provider it's all back to normal. Perhaps I used too big of a thought shortcut to explain this. Tho I have no idea why it does that at the stage of auth-refresh and not when there is attempt to login using this particular provider. Anyway, I assume the keys I am inputting are not the right keys, but I have no resources other than low quality youtube videos without sound lol. Nobody knows how to implement this

**Reply by ganigeorgiev** [reply]
Without a code sample showing how you are sending the request or at least some details how you've configured the OAuth2 app I don't think I'll be able to help you.

I've manually tested all OAuth2 providers (_the only exception is Apple_) and the implementation should work correctly unless Instagram has changed something with their APIs (which I doubt that this is the case).

Make sure to check that there are no typos/spaces in the client/app id and secret and also that you use the correct OAuth2 redirect url (Facebook/Instagram I remember requiring HTTPS so if you are testing locally you may need something like [ngrok](https://ngrok.com/download) to create a public address for your localhost).

**Reply by laszlo1337** [reply]
I doubt it's the fault of PocketBase, as it works with Google and Discord. So I am not really requiring your help as a creator of PB (thank you because it's amazing 🙏🏻) But my question was to the community - help with configuring that inbred creation from Meta and make it work. I just need login with instagram in my app that's all

**Comment by laszlo1337** [comment]
Gonna bump this thread since it got lost and no responses. Sorry!

Anybody had success loggin in using Instagram as auth provider? Can't find any good guides and their docs are non-comprehensible at all

**Reply by plantshopping** [reply]
Heya did you solve this?

**Reply by ganigeorgiev** [reply]
@plantshopping I can't speak for the author of the post but note that the current Instagram Basic Display API is being deprecated and will no longer work after December 4th ([Official Meta announcement](https://developers.facebook.com/blog/post/2024/09/04/update-on-instagram-basic-display-api/)).

For more details, you could read https://github.com/pocketbase/pocketbase/discussions/5652 and its related PR. The new provider is merged in the `develop` branch and it is available in v0.23.0-rc but note that it works only with business/creator Instagram accounts (aka. personal Instagram accounts can't use it).
